packages/cli/README.md

```diff
@@ -1,66 +1,478 @@
+
+We'll continue to use the `build` command as an example as we discuss each of these individually. You can find `build`'s source code in full [here ](https://github.com/redwoodjs/redwood/blob/master/packages/cli/src/commands/build.js).
+
+#### command
+
+`command` is a string that represents the command at the CLI and effectively maps it to the handler that does the work.
+
+The command for `build` is:
+
+```
+export const command = 'build [side..]'
 ```
 
-OK, OK, so this isn't really part of the RedwoodJS CLI, per se. But we felt it belonged here anyway!
+<!-- TODO -->
+`'build'`, the first word in the string, specifies the name of the command, and `'[side..]'` indicates that `build` takes an optional positional argument (named `side`--relevant for the `builder` and the `handler`). The dots (`..`) trailing `side` indicate that you can provide an array of strings (see [Variadic Positional Arguments](https://github.com/yargs/yargs/blob/master/docs/advanced.md#variadic-positional-arguments)), for example:
+
+<!-- The square brakcets indicate that the `build` takes an optional positional argument named `side` (relevant for `builder` and `handler`). The dots (`..`) trailing `side` indicate that you can provide an array of strings (see [Variadic Positional Arguments](https://github.com/yargs/yargs/blob/master/docs/advanced.md#variadic-positional-arguments)), for example: -->
 
-We create a new Redwood application by running the `yarn create redwood-app <project-dir>` command, where `<project-dir>` is the path of the to-be Redwood project.
+```
+yarn rw build api web
+```
 
-For example:
+You can use angle brackets (`<>`) to indicate that a positional argument is required. For example,
+
+
+```
+export const command = 'build <side>'
+```
+
+makes invoking `build` without any arguments, like
+
+```
+yarn rw build
+```
+
+fail. See [Positional Arguments](https://github.com/yargs/yargs/blob/master/docs/advanced.md#positional-arguments) for all your options when it comes specifying arguments for commands.
+
+#### description
+
+`description` is a string that's shown in the help output. `build`'s description is
+
+```
+export const description = 'Build for production'
+```
+
+And runnning `yarn rw build help` outputs:
+
+```
+rw build [side..]
+
+Build for production
+...
+```
+
+Other than that, there's not much more to it. You should of course make the description descriptive. And for consistency, don't punctuate the it. I.e., we prefer this
+
+```
+export const description = 'Build for production'
+```
+
+to this
+
+```
+export const description = 'Build for production.'
+```
+
+#### builder
+
+`builder` configures the positional arguments and options for the command. 
+
+While `builder` can be an object, the [positional argument api](https://yargs.js.org/docs/#api-positionalkey-opt) is only available if buidler's a function, so that's what we go with. We'll discuss more about builder being an object when we talk about testing.
```

Comment:
```suggestion
While `builder` can be an object, the [positional argument api](https://yargs.js.org/docs/#api-positionalkey-opt) is only available if builder's a function, so that's what we go with. We'll discuss more about builder being an object when we talk about testing.
```